manifests/charts/istio-control/istio-discovery/files/gen-istio.yaml

```diff
@@ -425,13 +425,7 @@ data:
             {{- end}}
             ]
         - name: ISTIO_META_APP_CONTAINERS
-          value: |-
-            [
-              {{- range $index, $container := .Spec.Containers }}
-                {{- if ne $index 0}},{{- end}}
-                {{ $container.Name }}
-              {{- end}}
-            ]
+          value: "{{- range $index, $container := .Spec.Containers }}{{- if ne $index 0}},{{- end}}{{ $container.Name }}{{- end}}"
```

Comment:
Can you try using sprig join function ?


http://masterminds.github.io/sprig/string_slice.html

May be it will not work,
In which case you can just define a new function, this level of templating control flow is just too much noise.

Reply:
I think we can make use of this in other variables too.. will try in a different PR..

Reply:
We might not use sprig here. If we don't we should add it